rewrite-java/src/main/java/org/openrewrite/java/JavaTemplate.java

```diff
@@ -417,7 +417,12 @@ public J visitMethodInvocation(J.MethodInvocation method, Integer integer) {
                     }
                     return m;
                 }
-                return maybeReplaceStatement(method, J.class, true, 0);
+                Cursor varDeclsCursor = getCursor().dropParentUntil(v -> v instanceof J.CompilationUnit
+                        || v instanceof J.VariableDeclarations
+                        || v instanceof J.Lambda
+                        || v instanceof J.ClassDeclaration
+                );
+                return maybeReplaceStatement(method, J.class, varDeclsCursor.getValue() instanceof J.VariableDeclarations, 0);
```

Comment:
Is this backwards? In your last commit weren't you setting this to `true` when it wasn't part of a variable declaration?

Reply:
yes, an oversight from trying different options and moving things around.
even corrected this test fails
```kotlin
recipe = toRecipe {
    object : JavaVisitor<ExecutionContext>() {
        val matcher = MethodMatcher("Integer valueOf(..)")
        val t = JavaTemplate.builder({ cursor }, "new Integer(#{any()})").doBeforeParseTemplate(System.out::println).build()
        override fun visitMethodInvocation(method: J.MethodInvocation, p: ExecutionContext): J {
            if (matcher.matches(method)) {
                return method.withTemplate(t, method.coordinates.replace(), method.arguments.get(0))
            }
            return super.visitMethodInvocation(method, p)
        }
    }
},
before = """
    import java.util.function.Function;
    class Test {
        Function<String, Integer> asInteger = it -> Integer.valueOf(it);
    }
""",
after = """
    import java.util.function.Function;
    class Test {
        Function<String, Integer> asInteger = it -> new Integer(it);
    }
"""
```